### ISSUE TITLE
Misleading error message when Cc field contains malformed address

### ISSUE THREAD

**Issue by ssoloff** [triplea-game/dice-server#33]
#### Reproduction steps

1. Start TripleA client.
1. Start a new PBEM/PBF game.
1. Enter a registered email in the Dice Server **To** field.
1. Enter a registered email in the Dice Server **Cc** field and add a trailing space.
1. Enter a value in the **Game ID** field.
1. Click **Test Server**.

#### Expected behavior

MARTI should indicate the address in the **Cc** field is not registered.

#### Actual behavior

MARTI indicates the address in the **To** field is not registered.

#### Notes

Originally from an axisandallies.org thread ending [here](https://www.axisandallies.org/forums/index.php?topic=41182.msg1729160#msg1729160).

**Comment by ssoloff** [comment]
After reviewing #32, I think that PR will address this issue.

**Comment by RoiEXLab** [comment]
@ssoloff I'm curious how exactly this issue will be resolved by my PR 😅
But anyways, good to hear ^^

**Comment by ssoloff** [comment]
@RoiEXLab #32 reports _all_ emails that weren't registered, whereas the current code only reports the first unregistered email.  However, I suspect there is a logic bug in this part of the current code:

```php
foreach($emails as $email) {
	if (!in_array($email, $registered_mails)) {
		exit("fatal error: email $email is not registered. Please register email at {$this->domain}/register.php !");
	}
}
```

that's causing MARTI to use a registered email in the error message.  Although I can't reason it out in my head at the moment. :smile:  I'm going to do some testing of #32 now anyway, so I'll verify if my assertion is true.

**Comment by ssoloff** [comment]
@RoiEXLab So after a bit of testing, I'll now say that #32 will partially address this problem.  Using the repro steps in this issue, MARTI no longer erroneously reports that a registered email is not registered.  Rather, it reports an _empty_ list of addresses as not being registered. 😕 

~It appears the root cause has to do with how MySQL does string matching.  Assume I have the registered address `foo@bar.com`.  If I run a query against that address with a trailing space, MySQL still returns a non-empty recordset:~

```
mysql> SELECT registered_email FROM dice_emails WHERE registered_email='foo@bar.com ';
+------------------+
| registered_email |
+------------------+
| foo@bar.com      |
+------------------+
1 row in set (0.00 sec)

mysql> SELECT registered_email FROM dice_emails WHERE registered_email IN ('foo@bar.com ');
+------------------+
| registered_email |
+------------------+
| foo@bar.com      |
+------------------+
1 row in set (0.00 sec)
```

~The problem arises when you then look for the difference between `$emails` and `$result` because PHP does not ignore the trailing space in the string comparison.~

~Apparently, the MySQL behavior is [correct according to the SQL standard](https://support.microsoft.com/en-us/help/316626/inf-how-sql-server-compares-strings-with-trailing-spaces).  So the PR for this issue will have to go a bit further than #32 to correct the problem.~

**EDIT:** The array of missing addresses is not actually empty--it contains one empty string.  See next comment.

**Comment by ssoloff** [comment]
After further investigation, the real root cause appears to be that we allow users to specify multiple email addresses in the To and Cc fields as space-separated lists.  When the user specifies a trailing (or presumably a leading) space, the `explode()` call in _MARTI.php_ adds an empty string to the final email list.  It's this empty email that is triggering the "not registered" error from MARTI.  It would probably be appropriate to trim the leading and trailing whitespace from the To and Cc values received from the client before calling `explode()`.

The root cause of the wrong email address being reported as not registered is due to incorrect usage of the `fetch_array()` function.  This function only returns information about a single record, and thus it should be called in a loop.  Therefore, `registered_emails` does not contain what the original author thought it contains, and we end up (in most cases) displaying the first element in `$emails` as the unregistered address.  Testing confirms that issue has been addressed by #32.
